tools/dune_rule_gen/gen_rules.ml

```diff
@@ -0,0 +1,29 @@
+open Coq_dune
+
+(* gen_rules theory_name dir *)
+let main () =
+  let tname = [Sys.argv.(1)] in
+  let fmt = Format.std_formatter in
+  let base_dir = Sys.argv.(2) in
+  let native = match Coq_config.native_compiler with
+    | Coq_config.NativeOff -> false
+    | Coq_config.NativeOn _ -> true
+  in
+  let root_lvl = List.length (String.split_on_char '/' base_dir) in
+  let boot = if tname = ["Coq"]
+    then Coq_rules.Boot_type.Stdlib
+    else Coq_rules.Boot_type.Regular (Some (Path.(adjust ~lvl:root_lvl (make "theories")))) in
+  Unix.chdir base_dir;
+  let vo_rules = Coq_rules.gen_vo_rules ~root_lvl ~tname ~native ~boot in
+  let package = base_dir in
+  Coq_rules.output_vo_and_install_rules fmt ~package vo_rules;
+  Format.pp_print_flush fmt ();
+  ()
+
+let () =
+  Printexc.record_backtrace true;
+  try main ()
+  with exn ->
+    let bt = Printexc.get_backtrace () in
+    let exn = Printexc.to_string exn in
+    Format.eprintf "[gen_rules] Fatal error: %s@\n%s@\n%!" exn bt
```

Comment:
Why does the tool return with exit code 0 in case of a fatal error?

Reply:
That's a bug, thanks!